WordPress/src/main/java/org/wordpress/android/ui/posts/PostListAction.kt

```diff
@@ -23,7 +39,49 @@ sealed class PostListAction {
     class DismissPendingNotification(val pushId: Int) : PostListAction()
 }
 
-fun handlePostListAction(activity: FragmentActivity, action: PostListAction) {
+private fun getPostsListStrategyFunctions(
+    activity: FragmentActivity,
+    action: PostListAction.PreviewPost
+) = object : RemotePreviewLogicHelper.RemotePreviewHelperFunctions {
+    override fun notifyNoNetwork() {
+        action.showSnackbar.invoke(action.messageNoNetwork)
+    }
+
+    override fun notifyUploadInProgress(post: PostModel): Boolean {
+        return if (UploadService.hasInProgressMediaUploadsForPost(post)) {
+            action.showToast.invoke(action.messageMediaUploading)
+            true
+        } else {
+            false
+        }
+    }
+
+    override fun startUploading(isRemoteAutoSave: Boolean, post: PostModel) {
+        if (isRemoteAutoSave) {
+            action.triggerPreviewStateUpdate.invoke(
+                    PostListRemotePreviewState.REMOTE_AUTO_SAVING_FOR_PREVIEW,
+                    PostInfoType.PostNoInfo
+            )
+            if (!UploadService.isPostUploadingOrQueued(post)) {
```

Comment:
I'd consider logging a message in the else statement. The method is called `startUploading` but it sometimes doesn't actually start any upload. Logging a message could help us debugging possible issues. Wdyt?

Reply:
👍 Done here 4a8bd66598bbd7b7179e3b37fdaf62dc02fafbd5.